glue-exp/importers/vizier/vizier.py

```diff
@@ -0,0 +1,77 @@
+# This plugin requires OpenCV to be installed, and returns a PIL Image object
+
+import os
+import sys
+from io import BytesIO
+
+import requests
+from astropy.io.votable import parse, parse_single_table
+
+from PyQt4.QtGui import QDialog, QApplication, QTreeWidgetItem, QHeaderView
+from PyQt4.uic import loadUi
+from PyQt4.QtCore import Qt
+
+UI_MAIN = os.path.join(os.path.dirname(__file__), 'vizier.ui')
+
+
+class VizierImporter(QDialog):
+
+    def __init__(self):
+
+        super(VizierImporter, self).__init__()
+
+        self.ui = loadUi(UI_MAIN, self)
+
+        self.cancel.clicked.connect(self.reject)
+        self.ok.clicked.connect(self.finalize)
+        self.search_button.clicked.connect(self.search)
+
+    def search(self):
+        text = self.query.text()
+        print("Searching " + text)
+        r = requests.post("http://vizier.u-strasbg.fr/viz-bin/votable", {'-words': text.split(), '-meta.all': 1})
+        print("Parsing VO Table")
+        self._checkboxes = {}
+        votablefile = parse(BytesIO(r.content))
+        for resource in votablefile.resources:
+            main = QTreeWidgetItem(self.tree.invisibleRootItem(), [resource.description, ""])
+            for table in resource.tables:
+                sub = QTreeWidgetItem(main)
+                sub.setFlags(sub.flags() | Qt.ItemIsUserCheckable)
+                sub.setCheckState(2, Qt.Unchecked)
+                sub.setText(0, table.description)
+                sub.setText(1, str(table.nrows))
+                sub.setText(2, "")
+                self._checkboxes[table.name] = sub
+        self.tree.resizeColumnToContents(0)
+        self.tree.resizeColumnToContents(1)
+        self.tree.resizeColumnToContents(2)
+
+    def finalize(self):
+
+        retrieve = []
+
+        for name in self._checkboxes:
+            if self._checkboxes[name].checkState(2) > 0:
+                retrieve.append(name)
+
+        self.tables = []
+
+        for iname, name in enumerate(retrieve):
+            print(iname / float(len(retrieve)) * 100.)
+            self.progress.setValue(iname / float(len(retrieve)) * 100.)
+            self.app.processEvents()
```

Comment:
It was a hack for now to get the progress bar to visually update. If you know of a better solution, let me know! (otherwise will look into it more over the weekend)